logger.go

```diff
@@ -75,24 +99,115 @@ func (l *MatchingLogger) Write(p []byte) (int, error) {
 	}
 
 	if mask {
+		// If the capture line matches any matcher and m.mask=true,
+		// don't republish it below.
 		return len(p), nil
 	}
 
-	// filter out already-timestamped logging from stdout
-	dateRegex := regexp.MustCompile(`^[0-9]{4}-[0-9]{2}-[0-9]{2}T`)
-	if dateRegex.Match(p) {
-		n, err := os.Stdout.Write(p)
-		if err != nil {
-			return n, err
-		}
-	} else {
+	// TODO(RSDK-7895): the lines from subprocess stdout are sometimes multi-line.
+	dateMatched := dateRegex.Match(p)
+	if !dateMatched {
+		// this case is the 'unstructured error' case; we were unable to parse a date.
 		lines := strings.ReplaceAll(strings.TrimSpace(string(p)), "\n", "\n\t")
+		entry := logging.LogEntry{Entry: zapcore.Entry{
+			Level:      zapcore.Level(logging.WARN),
+			Time:       time.Now().UTC(),
+			LoggerName: l.logger.Desugar().Name(),
+			Message:    fmt.Sprintf("unstructured output:\n\t%s", lines),
+			Caller:     zapcore.EntryCaller{Defined: false},
+		}}
 		if l.defaultError {
-			l.logger.Error(fmt.Sprintf("unstructured error output:\n\t%s", lines))
-		} else {
-			l.logger.Warn(fmt.Sprintf("unstructured output:\n\t%s", lines))
+			entry.Level = zapcore.Level(logging.ERROR)
+		}
+		// TODO(APP-5330): the l.logger.Write + globalNetAppender.Write here is because subloggers don't get netappenders
+		// that are added after the fact. Once this is fixed, revert to old approach.
+		l.logger.Write(&entry)
+		err := globalNetAppender.Write(entry.Entry, nil)
+		if err != nil {
+			fmt.Printf("error writing to NetAppender %s", err) //nolint:forbidigo
+		}
+	} else if l.uploadAll {
+		// in this case, date matching succeeded and we think this is a parseable log message.
+		// we check uploadAll because some subprocesses have their own netlogger which will
+		// upload structured logs. (But won't upload unmatched logs).
+		entry := parseLog(p).entry()
+		l.logger.Write(&logging.LogEntry{Entry: entry})
```

Comment:
This needs to ALWAYS happen, not just when uploadAll is set. The logger is writing the local log too, and that local log needs to always be written.